spec/Appendix B -- Grammar Summary.md

```diff
@@ -140,10 +140,14 @@ Selection :
   - FragmentSpread
   - InlineFragment
 
-Field : Alias? Name Arguments? Directives? SelectionSet?
+Field : Alias? Name Arguments? Nullability? Directives? SelectionSet?
```

Comment:
I don't have any strong opinions one way or the other. We went with the current ordering because that's what we're used to in some other languages.

Reply:
I don't have a strong opinion, but it would be helpful to dig up precedent across other languages

Reply:
Thinking about the null chaining operator in JS, `value?.(arg).prop` and `value(arg)?.prop` are both legal but mean different things. The first implies that `value` may or may not exist, and the second implies that the call to `value` may or may not return a value. It does seem like the second better matches our semantics where `field?(arg)` might imply that `field` might not be queryable (which doesn't make sense) where `field(arg)?` implies field might not return a value (which is exactly what we're talking about here)

Reply:
For Swift we've got something similar `value(arg)?.prop` would work the same as JS eg retrieve and return `prop` if the function call `value(arg)` doesn't return `null`, otherwise return `null`. 

`value?(arg)` would mean that the function `value` is optional, and should only execute `value` if it's not `null`, otherwise return `null`.

Reply:
Yeah I think I agree with your reasoning.